gapic/src/main/com/google/gapid/util/Logging.java

```diff
@@ -109,12 +121,12 @@ public static File getLogDir() {
     return logDir.get().isEmpty() ? null : new File(logDir.get());
   }
 
-  public static Log.Message[] getLogMessages() {
-    return buffer.getMessages();
+  public static Iterator<Log.Message> getMessageIterator() {
+    return buffer.iterator();
   }
 
-  public static void setListener(Runnable listener) {
-    buffer.listener = (listener == null) ? () -> { /* empty */ } : listener;
+  public static void setListener(Listener listener_) {
```

Comment:
Name shadowing is acceptable on setters, if you're shadowing the variable you're setting.

If you don't like to do that, please use new<Name> instead of <name>_.